gen/gen_mpy.py

```diff
@@ -2013,7 +2024,7 @@ def try_generate_array_type(type_ast):
         qualified_ptr_type = qualified_element_ptr_type,
         check_dim = '//TODO check dim!' if dim else '',
         mp_to_lv_convertor = mp_to_lv[element_type],
-        lv_to_mp_convertor = lv_to_mp[element_type],
+        lv_to_mp_convertor = lv_to_mp[element_type_ptr] if struct_type and dim else lv_to_mp[element_type],
```

Comment:
`try_generate_array_type` is used in a wider context. For example, when returning a struct as a return value from a function. In such case, it's wrong to pass the struct by reference.  
So I suspect that just updating `lv_to_mp_convertor` is too general and could be wrong in certain cases.

Reply:
see the reply in the previous comment and you will see that the only thing that are effected is a really small list and what is being changed are the things that do need to be changed.